### ISSUE TITLE
Error message on creating simple maps which worked yesterday.

### ISSUE THREAD

**Issue by SwultCode** [dkahle/ggmap#353]
## number of rows of matrices must match

Hey, `ggmap` was working just fine yesterday, when I used it today I couldn't make any large maps, by large I mean a big bounding box, the zoom isn't the issue. Whenever I make a map, i.e. one of europe, I get an error of `Error in f(init, x[[i]]) : 
  number of rows of matrices must match (see arg 2)`. When I change the bounding box for `ggmap` to be smaller it works, so I assume it's an issue somewhere there.

Below is a reproduce-able example of the error, it continues to happen after I restart r and my computer.

```r
library(ggmap)
lonlatTest = data.frame(lon=c(-40,15), lat=c(30,0))
qmplot(lon,lat,data=lonlatTest,zoom=5)
```
`Error in f(init, x[[i]]) : 
  number of rows of matrices must match (see arg 2)`

**Comment by xp-song** [comment]
I've also experienced this in the function `get_map()`. I input a bounding box and set `source = "stamen"`. The error traceback seems to suggest that it happens within the `stitch()` function in `get_stamenmap()`
![Screenshot 2023-10-04 at 15 36 43](https://github.com/dkahle/ggmap/assets/36692010/63420a43-53e7-4a96-ab08-f4880121da38)

**Comment by ryurko** [comment]
I'm experiencing the exact same error for code that used to work:
`library(ggmap)`
`US <- c(left = -125, bottom = 10,  right = -67, top = 49)`
`map <- get_stamenmap(US, zoom = 5, maptype = "toner-lite")`

**Comment by ZacWarham** [comment]
I had the same issue but after some digging it turns out that [Stamen Maps no longer hosts tiles](https://maps.stamen.com/stadia-partnership/) - manually creating a URL and visiting shows service unavailable. This would require a lot of reworking to use a third-party cache of the Stamen Maps or trying to link to the API for the new host similar to how Google Maps requires an API key

**Comment by ZacWarham** [comment]
> I had the same issue but after some digging it turns out that [Stamen Maps no longer hosts tiles](https://maps.stamen.com/stadia-partnership/) - manually creating a URL and visiting shows service unavailable. This would require a lot of reworking to use a third-party cache of the Stamen Maps or trying to link to the API for the new host similar to how Google Maps requires an API key

Turns out the team at StadiaMaps created a PR for this a while ago https://github.com/dkahle/ggmap/pull/351 which adds a `register_stadiamaps()` function and replaces `get_stamenmap` with `get_stadiamap`

Current workaround until @dkahle merges the PR (skip step 3 if reading this after merge and just update your install instead)

1. Create account for [Stadia Maps](uhttps://stadiamaps.com/stamen/onboarding/create-account/rl) (they have a free option but depends on your use case)
2. Create a 'Property' on your account and get an API key
3. Install this fork https://github.com/stadiamaps/ggmap
- You may need to do this on Windows like I did https://github.com/forestgeo/learn/issues/102#issuecomment-1379777330
4. register your API key using `register_stadimaps()`
5. Replace all instances of `get_stamenmap` with `get_stadiamap`
6. Make sure to change the tile type to the new format (there will be an error if incorrect type)

**Comment by aradzikbys** [comment]
Regarding point 3 >> details on re-instalation of ggmap can be found here: https://docs.stadiamaps.com/guides/migrating-from-stamen-map-tiles/#ggmap

**Comment by dkahle** [comment]
I believe this is fixed by 191159d; please reopen if that isn't the case.
